modules/connectivity/TrackStreamingStatus.ts

```diff
@@ -0,0 +1,651 @@
+import { getLogger } from '@jitsi/logger';
+
+import * as JitsiConferenceEvents from '../../JitsiConferenceEvents';
+import * as JitsiTrackEvents from '../../JitsiTrackEvents';
+import RTCEvents from '../../service/RTC/RTCEvents';
+import { createTrackStreamingStatusEvent } from '../../service/statistics/AnalyticsEvents';
+import JitsiConference from '../../types/hand-crafted/JitsiConference';
+import JitsiRemoteTrack from '../../types/hand-crafted/modules/RTC/JitsiRemoteTrack';
+import RTC from '../../types/hand-crafted/modules/RTC/RTC';
+import browser from '../browser';
+import Statistics from '../statistics/statistics';
+
+/** Track streaming statuses. */
+type TrackStreamingStatus = 'active' | 'inactive' | 'interrupted' | 'restoring';
+interface TrackStreamingStatusMapType {
+    [key: string]: TrackStreamingStatus
+}
+
+type VideoType = 'camera' | 'desktop';
+type StreamingStatusMap = {
+    videoType?: VideoType,
+    startedMs?: number,
+    p2p?: boolean,
+    streamingStatus?: string,
+    value?: number
+};
+
+const logger = getLogger(__filename);
+
+/**
+ * Default value of 500 milliseconds for {@link TrackStreamingStatusImpl.outOfForwardedSourcesTimeout}.
+ */
+const DEFAULT_NOT_IN_FORWARDED_SOURCES_TIMEOUT = 500;
+
+/**
+ * Default value of 2500 milliseconds for {@link TrackStreamingStatusImpl.p2pRtcMuteTimeout}.
+ */
+const DEFAULT_P2P_RTC_MUTE_TIMEOUT = 2500;
+
+/**
+ * Default value of 10000 milliseconds for {@link TrackStreamingStatusImpl.rtcMuteTimeout}.
+ */
+const DEFAULT_RTC_MUTE_TIMEOUT = 10000;
+
+/**
+ * The time to wait a track to be restored. Track which was out of forwarded sources should be inactive and when
+ * entering forwarded sources it becomes restoring and when data is received from bridge it will become active, but if
+ * no data is received for some time we set status of that track streaming to interrupted.
+ */
+const DEFAULT_RESTORING_TIMEOUT = 10000;
+
+export const TrackStreamingStatusMap: TrackStreamingStatusMapType = {
```

Comment:
I'll take another look into this. I think I was having issues when I first tried it because `TrackStreamingStatusMap` was being imported into non `.ts` files where `enum`s aren't supported